iothub/device/src/Transport/Amqp/AmqpTransportHandler.cs

```diff
@@ -116,10 +143,10 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
                 _closed = true;
             }
 
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Why is this being moved outside of the try block?

Reply:
We chose to do that in public API surface. Not in internal methods?

Reply:
Copying the pattern we had in service cleint

Reply:
reverted it